### PR TITLE
Syndicate Objective Manifestation System (SOMS)

### PR DESCRIPTION
**Pull request by ynot01**
# Document the changes in your pull request

Adds a new category to the uplink with 4 options:

- Request Objective (RANDOM)
- Request Objective (EASY)
- Request Objective (MEDIUM)
- Request Objective (HARD)

Easy objectives will reward 2-3 TC
Medium objectives will reward 3-5 TC
Hard objectives will reward 5-7 TC

<details>
  <summary>Easy Objectives</summary>

- Kill a pet
- Steal 28 moles of plasma (full tank)
- Steal an unused slime extract

</details>

<details>
  <summary>Medium Objectives</summary>

- Kill someone once
- Steal a reflective jacket
- Steal a hand teleporter
- Steal CE magboots
- Steal a medal of captaincy
- Steal the hypospray
- Steal reactive tele armor
- Steal secret documents

</details>

<details>
  <summary>Hard Objectives</summary>

- Kill an AI
- Steal antique laser gun
- Steal HoS gun
- Steal captain's jetpack
- Steal nuke disk
- Steal an AI
- Steal the blueprints
- Steal the blackbox

</details>

Each option has a 50% chance of appearing/not appearing and only have one use.

"Purchasing" one of these will notify online admins and start a 1 minute timer.

If an admin is online, they will have the option to override your objective with their own custom objective, and they will be able to choose their own difficulty rating.

Once the 1 minute timer expires or an admin creates an objective, you will be notified.

Your objective will come in form of a folder of a random color on a random safe turf in the station where no one can currently see it. It is up to you to get to it.

You will be told where and what color the folder is.

If another syndicate gets ahold of your folder, they will be able to complete the objective just as you would have. You can use this to steal or cooperate with others.

Once you have completed the objective, you should hit the Check Completion button. 

If the objective is complete, the folder will give you your reward and add the objective to your objectives list (so you don't abandon it after).

If the objective is incomplete, the folder will scold you.

If the objective is admin-handled, admins will be notified of your request and either confirm or deny the action. You will be alerted to their action by a beep and if they confirm, the objective will be marked as complete and you will be able to redeem the objective as normal.

## Visual Aid

image

1 minute later...

# Changelog

:cl:  
experimental: Added objective requesting system for Syndicates
bugfix: Fixed a bunch of objectives runtiming if they couldn't find what they wanted
tweak: Made caplaser objective 10 difficulty from 5
tweak: Made captain's jetpack objective 10 difficulty from 5
tweak: Made nuke disk objective 15 difficulty from 5
tweak: Made reflective vest objective 5 difficulty from 3
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
oooo

**Comment by 00ze-cyclone** [comment]
This sound interesting.

**Comment by ghost** [comment]
Assasination should count as a medium objective

**Comment by Marmio64** [comment]
this is a cool idea but im not so sure on the balance implications. Doing 1 medium/hard mission can allow you to easily cross the TC threshold for a lot of things that are otherwise very difficult to get. You'd need to be an IAA and kill some targets, gamble all your TC on being a contractor, or manage to link up and borrow TC from another traitor (or steal it). This however seems to have virtually no risk and if the given objective is an easy one to get for you in the current round, it can be very powerful. If you don't like the objective, as far as I know you can just choose to completely ignore it. Even though you wont get any extra TC you wont specifically lose anything from not going with it. I think a better idea would be to integrate this with the contractor stuff thats already in place since it has much of the same idea anyways and would keep it so that you actually risk stuff for extra TC.

TL;DR
you should just integrate this with the contractor system instead

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -2314,3 +2315,63 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
+	difficulty = 2
+
+/datum/uplink_item/new_objective/hard
+	name = "Request Objective (HARD)"
+	difficulty = 3
+
+/datum/uplink_item/new_objective/spawn_item(spawn_path, mob/user, datum/component/uplink/U)
+	to_chat(user, span_notice("Signal sent to command. Awaiting response (ETA ~1 minute)..."))
+	var/diff_txt = list("RANDOM", "EASY", "MEDIUM", "HARD")[difficulty+1]
+	message_admins("[ADMIN_LOOKUPFLW(user)] has requested an objective ([diff_txt]). (<A HREF='?_src_=holder;[HrefToken()];uplink_custom_obj=[REF(src)];requester=[REF(user)]'>FORGE CUSTOM OBJECTIVE?</A>) (AUTO SET IN 1 MINUTE)")
```
Comment (redmoogle) [inline code/modules/uplink/uplink_items.dm:3514]: What about a deny button for when the round is already too chaotic

**Comment by ynot01** [comment]
> you should just integrate this with the contractor system instead

**Comment by GDRdev** [comment]
### **this surely will effect the tc economy**

Though imagine the anoumts of tc you could get in one game...
Contactor, syndie bounties, and now THIS?

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
can you make the button in the uplink say "REQUEST" instead of 0 tc?

**Comment by GDRdev** [comment]
There is also reason to not get a extra hard objective. Since it doesnt effect your _greentext_ failling it would just mean you dont get the **extra** tc. If I had to suggest a change it would probably make it soo you have to spend like a down payment of tc to get these objectives soo you do lose out on something when you fail it instead of it being extra tc for free.

**Comment by ynot01** [comment]
> Since it doesnt effect your _greentext_ failling

If you complete the objective (and receive your reward TC) the objective is added to your objectives list to prevent fraud

**Comment by GDRdev** [comment]
also how is the "Steal a reflective jacket" a medium objective when - nvm you can just raid cargo-

Though there is no reason why the blueprints is a **"hard"** objective considering its just as hard as the boots if not easier since CE normally doesnt carry around the blueprints unlike the boots.

**Comment by GDRdev** [comment]
> > Since it doesnt effect your _greentext_ failling
> 
> If you complete the objective (and receive your reward TC) the objective is added to your objectives list to prevent fraud

You know what would be better than that? If you were given a certain amount of time to complete the objective and if you fail a single IIA would spawn with the only objective to kill you.

**Comment by ynot01** [comment]
> also how is the "Steal a reflective jacket" a medium objective when - nvm you can just raid cargo-

The easy objectives are *incredibly* easy and I do not believe the reflective jacket is on the same level.

**Comment by GDRdev** [comment]
> > Since it doesnt effect your _greentext_ failling
> 
> If you complete the objective (and receive your reward TC) the objective is added to your objectives list to prevent fraud

This also doesnt change my point of you could just garb a extra hard objective and if its too hard you can just leave it and ignore the extra tc

**Comment by ynot01** [comment]
> This also doesnt change my point of you could just garb a extra hard objective and if its too hard you can just leave it and ignore the extra tc

I don't see an issue with allowing players to turn down objectives if they deem them too difficult

**Comment by GDRdev** [comment]
> > also how is the "Steal a reflective jacket" a medium objective when - nvm you can just raid cargo-
> 
> The easy objectives are _incredibly_ easy and I do not believe the reflective jacket is on the same level.

I was talking about moving it up to hard since you would normally have to raid armory (and assuming this is on high pop) is very hard to do but I remembered you can just order it

**Comment by GDRdev** [comment]
> > This also doesnt change my point of you could just garb a extra hard objective and if its too hard you can just leave it and ignore the extra tc
> 
> I don't see an issue with allowing players to turn down objectives if they deem them too difficult

*shrug I guess the only issue is that it could be too free of a objective which allows you to start with 27 tc instead of the normal 20. Sorta like syndie bounties on low pop

**Comment by ynot01** [comment]
> *shrug I guess the only issue is that it could be too free of a objective which allows you to start with 27 tc instead of the normal 20.

That's why the objectives are sorted by difficulty. All of the 'hard' objectives are very respectably difficult objectives.

**Comment by GDRdev** [comment]
> > *shrug I guess the only issue is that it could be too free of a objective which allows you to start with 27 tc instead of the normal 20.
> 
> That's why the objectives are sorted by difficulty. All of the 'hard' objectives are very respectably difficult objectives.

still dont get blueprints as hard ngl

**Comment by MajManatee** [comment]
Seems like too much TC imo

**Comment by cowbot92** [comment]
I like this

**Comment by Therandomhoboo** [comment]
If I'm reading this right, a potential 12-15 tc added on if you get all folders with good rng.
This just looks like bounties but instead of having to use an emag you just have to gamble but everyone has it. At least with bounties it was locked behind cargo or you needed to build a console to do it and use the supply shuttle to send it off so they had to be stuck to one area. 

I'm just not a fan of it because bounties exist currently and that the risk seems kinda small unless you roll extremely bad.

**Review by ToasterBiome — APPROVED** [review]

**Comment by Moltijoe** [comment]
Balance Council:
>We do not currently want a system like this in the game

**Comment by ToasterBiome** [comment]
HD reopen, testmerging

**Comment by Maia-J** [comment]
No way yog progtot

**Comment by LassiH** [comment]
overpowered free tc reskinned syndicate bounties for free no emag purchase required
it is just contractors but free if syndicate bounties got removed why are we adding more ways for traitors to get more than  20 tc without help from others

**Comment by Mqiib** [comment]
Folders can spawn in non-station areas like the syndicate research outpost on icemoon. It happened twice in one round to @Yarinoi 

image
image

**Comment by Yarinoi** [comment]
can confirm it happened to me
image
image

**Review by Majkl-J — CHANGES_REQUESTED** [review]
Spawns should be limited to `/area/station` subtypes

**Comment by ynot01** [comment]
to be completed at a later date
